docs/what-it-does.md

```diff
@@ -0,0 +1,103 @@
+# What the Consul Deployment Agent Does
+
+## On startup
+
+- Waits for puppet to drop file to signal instance readiness before proceeding.
+
+- Loads consul and standard configuration from the local config.yml file.
+
+- Loads server role information from the host EC2 instance's tags (or hard-coded values during local development).
+
+- Figures out server role key and constantly polls (long polling) for changes in key value store. When a change is detected the convergence process is run (see [Convergence](#convergence) process below). Key format is... `environments/{ENVIRONMENT}/roles/{SERVER_ROLE}`
+
+<a id="convergence"></a>
+
+## Convergence
+
+- Loads list of serviceSlices registered in the catalog.
+
+- Loads list of serviceSlices in KV store.
+
+	- Key to list services: `environments/{ENVIRONMENT}/roles/{SERVER_ROLE}/services`
+	- Key to list slices for a service: `environments/{ENVIRONMENT}/roles/{SERVER_ROLE}/services/{SERVICE_NAME}`
+
+- For each serviceSlice listed...
+
+  - pulls details from: `environments/{ENVIRONMENT}/roles/{SERVER_ROLE}/services/{SERVICE_NAME}/{SLICE}`. This gives us the deploymentId and version number. We then collect further details from:
+    - `environments/{ENVIRONMENT}/services/{SERVICE_NAME}/{VERSION}/definition`
+    - `environments/{ENVIRONMENT}/services/{SERVICE_NAME}/{VERSION}/installation`
+    - `deployments/{DEPLOYMENT_ID}`
+  - Figures out which serviceSlices are not present or have the wrong version.
+  - Filters serviceSlices which are in 'Ignore' state or which have beenpreviously quarantened.
```

Comment:
What marks a service slice as ignore or quranteen? How do you make it so that the service *will* deploy in the future?